java/ql/src/Telemetry/AutomodelApplicationModeCharacteristics.qll

```diff
@@ -324,14 +325,15 @@ private class NonPublicMethodCharacteristic extends CharacteristicsImpl::Uninter
 private class OtherArgumentToModeledMethodCharacteristic extends CharacteristicsImpl::LikelyNotASinkCharacteristic
 {
   OtherArgumentToModeledMethodCharacteristic() {
-    this = "other argument to a method that has already been modeled"
+    this = "other argument to a method that has already been modeled manually"
   }
 
   override predicate appliesToEndpoint(Endpoint e) {
-    not ApplicationCandidatesImpl::isSink(e, _) and
-    exists(DataFlow::Node otherSink |
-      ApplicationCandidatesImpl::isSink(otherSink, _) and
-      e.asExpr() = otherSink.asExpr().(Argument).getCall().getAnArgument() and
+    not ApplicationCandidatesImpl::isSink(e, _, _) and
+    exists(DataFlow::Node otherSink, Call c |
+      ApplicationCandidatesImpl::isSink(otherSink, _, "manual") and
+      c = otherSink.asExpr().(Argument).getCall() and
+      e.asExpr() in [c.getQualifier(), c.getAnArgument()] and
```

Comment:
This isn't going to always work if `otherSink` is an instance argument (`this`), or if either `e` or `otherSink` are variadic arguments. I think we could take advantage of `ArgumentNode` here:

```suggestion
      e.getCall() = otherSink.(ArgumentNode).getCall()
```

This implies that `Call c` isn't needed at the `exists`, and that we add a `getCall` predicate to `ArgumentNode`. If we also considered `ImplicitVarargsArray` as part of the definition of `ArgumentNode` it would be the cherry on top:

```ql
private class ArgumentNode extends DataFlow::Node {
  Call c;

  ArgumentNode() {
    exists(Argument arg | this.asExpr() = arg and not arg.isVararg() and c = arg.getCall())
    or
    this.(DataFlow::ImplicitVarargsArray).getCall() = c
    or
    this.(DataFlow::InstanceAccessNode) = DataFlow::getInstanceArgument(c)
  }

  Call getCall() { result = c }
}
```

Reply:
[13027a1](https://github.com/github/codeql/pull/13747/commits/13027a1094ba9cf28a08eca514d7161ad8b6111d) I think this commit implements your suggestion :) thank you :) :)